tree/ntuple/v7/src/RMiniFile.cxx

```diff
@@ -1105,12 +1105,27 @@ ROOT::Experimental::Internal::RNTupleFileWriter::Recreate(std::string_view ntupl
    return writer;
 }
 
+std::pair<std::string_view, std::string_view> ROOT::Experimental::Internal::RNTupleFileWriter::SplitNTupleName(
+   std::string_view nameAndDir)
+{
+   const auto posLastSlash = nameAndDir.find_last_of('/');
+   if (posLastSlash == std::string::npos)
+      return std::make_pair("", nameAndDir);
+
+   if (posLastSlash == nameAndDir.size() - 1)
+      throw RException(R__FAIL("empty RNTuple name"));
+
+   return std::make_pair(nameAndDir.substr(0, posLastSlash), nameAndDir.substr(posLastSlash + 1));
+}
+
 std::unique_ptr<ROOT::Experimental::Internal::RNTupleFileWriter>
 ROOT::Experimental::Internal::RNTupleFileWriter::Append(std::string_view ntupleName, TFile &file,
                                                         std::uint64_t maxKeySize)
 {
-   auto writer = std::unique_ptr<RNTupleFileWriter>(new RNTupleFileWriter(ntupleName, maxKeySize));
+   auto [dir, name] = SplitNTupleName(ntupleName);
```

Comment:
Note that it would also be nice to have:
```
ROOT::Experimental::Internal::RNTupleFileWriter::Append(std::string_view ntupleName, TDIrectory &dir,
                                                        std::uint64_t maxKeySize) 
{
     if (!dir.GetFile())
        throw "We need a file";
   auto writer = std::unique_ptr<RNTupleFileWriter>(new RNTupleFileWriter(name, maxKeySize));
   writer->fFileProper.fFile = dir->GetFile();
   writer->fFileProper.fDirectory = &dir; 
```
(oups the type of `fDirectory` is wrong.